Refactor CoreDNS service (#453)
k8gb-io/k8gb: docs/exposing_dns.md

[Local project setup](./local.md) does this patching automatically.

## CoreDNS service lookup

> k8gb no longer hardcodes coredns LB service name, instead it is trying to find a service annotated with `app.kubernetes.io/name=coredns` within the same namespace where controller itself is deployed into. If service has `Status.LoadBalancer.Ingress[0].Hostname`, k8gb will use the resolved IPs to reach CoreDNS on this cluster.

Reviewer: I think we should not overload the documentation with the historical approach( which is good for commit message but not a doc). 
```suggestion
k8gb is trying to find a service annotated with `app.kubernetes.io/name=coredns` within the same namespace where the controller itself is deployed into. If the service has `Status.LoadBalancer.Ingress[0].Hostname`, k8gb will use the resolved IPs to reach CoreDNS on this cluster.
```

Author: fixed

> k8gb controller will first try to find CoreDNS, by looking into a namespace where controller is deployed and find service which annotated with `app.kubernetes.io/name=coredns`. If service have `Status.LoadBalancer.Ingress[0].Hostname` k8gb will use those IPs to reach CoreDNS on this cluster. In case service is not found or have `Status.Loadbalancer.Ingress[0].Hostname` empty, k8gb controller will fallback into Ingress IP lookup.

Reviewer 2: ```suggestion
k8gb controller will first try to find CoreDNS, by looking into a namespace where controller is deployed and finding a service annotated with `app.kubernetes.io/name=coredns`. If service has `Status.LoadBalancer.Ingress[0].Hostname`, k8gb will use the resolved IPs to reach CoreDNS on this cluster. In case is service is not found or its `Status.Loadbalancer.Ingress[0].Hostname` is empty, k8gb controller will fallback into Ingress IP lookup.
```

Reviewer 2: @k0da this suggestion doesn't look like resolved too.

> k8gb controller will first try to find CoreDNS, by looking into a namespace where controller is deployed and finding a service annotated with `app.kubernetes.io/name=coredns`. If service has `Status.LoadBalancer.Ingress[0].Hostname`, k8gb will use the resolved IPs to reach CoreDNS on this cluster. In case is service is not found or its `Status.Loadbalancer.Ingress[0].Hostname` is empty, k8gb controller will fallback into Ingress IP lookup.

Author: I need to get this statement to be updated. It is no longer a fallback. Behaviour stays the same, it just COREDNS_EXPOSED based on a condition if service type is LB
